File: js/maxchar.js
function maxChar(str) {
  const chars = {};
  let max = 0;
  let maxChar = '';
  for (let char of str){
    if (chars[char]){
      chars[char]++;
    }
    else{
      chars[char] = 1;
    }
  }

  for (let char in chars){
    if (chars[char] > max){
      max = chars[char];
      maxChar = char;
    }
  }
  return maxChar;
}

File: python/maxchar.py
##find a max character in a string
def maxchar(string):
    result = {}
    if not string:
        return ''
    for i in string:
        if i in result:
            result[i] += 1
        else:
            result[i] = 1
    return [x for (x,y) in result.items() if y == max(result.values())].pop()

print(maxchar("abcaabc"))
print(maxchar(""))